CUDA Tests broken in Develop

What are the exact, minimal CMake parameters to run the tests *manually* with CUDA enabled?

I am trying
```bash
mkdir build
cd build

cmake .. -DALPAKA_ACC_GPU_CUDA_ENABLE=ON -DALPAKA_CUDA_ARCH="37"
# ok

make -j
# ok

ctest -V
# fail 9/26
```

on a K80 but many tests fail:
```
65% tests passed, 9 tests failed out of 26

Total Test time (real) = 112.39 sec

The following tests FAILED:
	 12 - atomic (Failed)
	 13 - blockShared (Failed)
	 14 - blockSync (Failed)
	 18 - kernel (Failed)
	 19 - memBuf (Failed)
	 20 - memView (Failed)
	 21 - memP2P (Failed)
	 24 - rand (Failed)
	 25 - time (Failed)
Errors while running CTest
make: *** [test] Error 8
```


e.g. `rand`:
```
$ ./test/unit/rand/rand 
Running 18 test cases...
rand: /home/huebl/src/alpaka/include/alpaka/workdiv/WorkDivHelpers.hpp:192: alpaka::workdiv::WorkDivMembers<TDim, TIdx> alpaka::workdiv::subDivideGridElems(const alpaka::vec::Vec<TDim, TVal>&, alpaka::vec::Vec<TDim, TVal>, const alpaka::acc::AccDevProps<TDim, TIdx>&, bool, alpaka::workdiv::GridBlockExtentSubDivRestrictions) [with TDim = std::integral_constant<long unsigned int, 1ul>; TIdx = short int]: Assertion `isValidAccDevProps(accDevProps)' failed.
unknown location(0): fatal error: in "rand_/defaultRandomGeneratorIsWorking<N6alpaka3acc12AccGpuCudaRtISt17integral_constantImLm1EEsEE>": signal: SIGABRT (application abort requested)
/home/huebl/src/alpaka/test/unit/rand/src//RandTest.cpp(152): last checkpoint: "defaultRandomGeneratorIsWorking" entry.
rand: /home/huebl/src/alpaka/include/alpaka/workdiv/WorkDivHelpers.hpp:192: alpaka::workdiv::WorkDivMembers<TDim, TIdx> alpaka::workdiv::subDivideGridElems(const alpaka::vec::Vec<TDim, TVal>&, alpaka::vec::Vec<TDim, TVal>, const alpaka::acc::AccDevProps<TDim, TIdx>&, bool, alpaka::workdiv::GridBlockExtentSubDivRestrictions) [with TDim = std::integral_constant<long unsigned int, 2ul>; TIdx = short int]: Assertion `isValidAccDevProps(accDevProps)' failed.
unknown location(0): fatal error: in "rand_/defaultRandomGeneratorIsWorking<N6alpaka3acc12AccGpuCudaRtISt17integral_constantImLm2EEsEE>": signal: SIGABRT (application abort requested)
/home/huebl/src/alpaka/test/unit/rand/src//RandTest.cpp(152): last checkpoint: "defaultRandomGeneratorIsWorking" entry.
rand: /home/huebl/src/alpaka/include/alpaka/workdiv/WorkDivHelpers.hpp:192: alpaka::workdiv::WorkDivMembers<TDim, TIdx> alpaka::workdiv::subDivideGridElems(const alpaka::vec::Vec<TDim, TVal>&, alpaka::vec::Vec<TDim, TVal>, const alpaka::acc::AccDevProps<TDim, TIdx>&, bool, alpaka::workdiv::GridBlockExtentSubDivRestrictions) [with TDim = std::integral_constant<long unsigned int, 3ul>; TIdx = short int]: Assertion `isValidAccDevProps(accDevProps)' failed.
unknown location(0): fatal error: in "rand_/defaultRandomGeneratorIsWorking<N6alpaka3acc12AccGpuCudaRtISt17integral_constantImLm3EEsEE>": signal: SIGABRT (application abort requested)
/home/huebl/src/alpaka/test/unit/rand/src//RandTest.cpp(152): last checkpoint: "defaultRandomGeneratorIsWorking" entry.

*** 3 failures are detected in the test module "rand"
```

with CUDA 8.0.44, Boost 1.62 on Hypnos (HZDR) and `develop` as of 8e206bdad75da2704bdac7060a61115a97304df2

--- a/test/unit/mem/buf/src/BufTest.cpp
+++ b/test/unit/mem/buf/src/BufTest.cpp
@@ -63,7 +63,7 @@ struct CreateExtentBufVal
         TIdx)
     -> TIdx
     {
-        return sizeof(TIdx) * (5u - Tidx);
+        return static_cast<TIdx>(5u - Tidx);
     }
 };
 

--- a/test/unit/mem/p2p/src/P2P.cpp
+++ b/test/unit/mem/p2p/src/P2P.cpp
@@ -65,7 +65,7 @@ struct CreateExtentBufVal
         TIdx)
     -> TIdx
     {
-        return sizeof(TIdx) * (5u - Tidx);
+        return static_cast<TIdx>(5u - Tidx);
     }
 };
 

--- a/test/unit/mem/view/src/ViewSubViewTest.cpp
+++ b/test/unit/mem/view/src/ViewSubViewTest.cpp
@@ -80,7 +80,7 @@ namespace view
             TIdx)
         -> TIdx
         {
-            return sizeof(TIdx) * (5u - Tidx);
+            return static_cast<TIdx>(5u - Tidx);
         }
     };
 
@@ -100,7 +100,7 @@ namespace view
             TIdx)
         -> TIdx
         {
-            return sizeof(TIdx) * (4u - Tidx);
+            return static_cast<TIdx>(4u - Tidx);
         }
     };
 
@@ -242,7 +242,7 @@ namespace view
         auto buf(alpaka::mem::buf::alloc<TElem, Idx>(dev, extentBuf));
 
         auto const extentView(alpaka::vec::createVecFromIndexedFnWorkaround<Dim, Idx, CreateExtentViewVal>(Idx()));
-        auto const offsetView(alpaka::vec::Vec<Dim, Idx>::all(sizeof(Idx)));
+        auto const offsetView(alpaka::vec::Vec<Dim, Idx>::all(static_cast<Idx>(1)));
         View view(buf, extentView, offsetView);
 
         alpaka::test::mem::view::testViewSubViewMutable<TAcc>(view, buf, dev, extentView, offsetView);
@@ -268,7 +268,7 @@ namespace view
         auto buf(alpaka::mem::buf::alloc<TElem, Idx>(dev, extentBuf));
 
         auto const extentView(alpaka::vec::createVecFromIndexedFnWorkaround<Dim, Idx, CreateExtentViewVal>(Idx()));
-        auto const offsetView(alpaka::vec::Vec<Dim, Idx>::all(sizeof(Idx)));
+        auto const offsetView(alpaka::vec::Vec<Dim, Idx>::all(static_cast<Idx>(1)));
         View const view(buf, extentView, offsetView);
 
         alpaka::test::mem::view::testViewSubViewImmutable<TAcc>(view, buf, dev, extentView, offsetView);

